src/OrbitLinuxTracing/Function.h

```diff
@@ -5,28 +5,35 @@
 #ifndef ORBIT_LINUX_TRACING_FUNCTION_H_
 #define ORBIT_LINUX_TRACING_FUNCTION_H_
 
+#include <absl/hash/hash.h>
+
 #include <cstdint>
 #include <string>
 
 namespace orbit_linux_tracing {
 class Function {
  public:
-  Function(std::string binary_path, uint64_t file_offset, uint64_t virtual_address)
-      : binary_path_{std::move(binary_path)},
-        file_offset_{file_offset},
-        virtual_address_{virtual_address} {}
-
-  const std::string& BinaryPath() const { return binary_path_; }
+  Function(uint64_t id, std::string file_path, uint64_t file_offset)
+      : id_(id), file_path_{std::move(file_path)}, file_offset_{file_offset} {}
 
-  uint64_t FileOffset() const { return file_offset_; }
+  [[nodiscard]] uint64_t id() const { return id_; }
+  [[nodiscard]] const std::string& file_path() const { return file_path_; }
+  [[nodiscard]] uint64_t file_offset() const { return file_offset_; }
 
-  uint64_t VirtualAddress() const { return virtual_address_; }
+  bool operator==(const Function& other) {
+    return (this->file_offset_ == other.file_offset_) && (this->file_path_ == other.file_path_);
+  }
 
  private:
-  std::string binary_path_;
+  uint64_t id_;
```

Comment:
I think I would prefer `function_id_`. This is not really an id for this specific class, it's an id for the more general "concept" of "function" across the codebase.

Reply:
done.